persist/fs/commitlog/iterator.go

```diff
@@ -126,10 +126,7 @@ func (i *iterator) Close() {
 		return
 	}
 	i.closed = true
-	if i.reader != nil {
-		i.reader.Close()
-		i.reader = nil
-	}
+	i.closeAndResetReader()
```

Comment:
same here

Reply:
cool I made this one return false if there is an error if thats ok

Reply:
looks good. do you need to do the `closeErr` check on line 137 too?

Reply:
the current implementation of the Close() method doesn't return an error so there is nowhere to bubble it up to

Reply:
I could leave a TODO with a note to refactor the rest of the codebase to support an error on Close()?

Reply:
Yep TODO sounds good

Reply:
+1 for the todo